starters/express-typescript-typeorm-postgres/src/utils/log-helper.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const allowedLogLevels = process.env.ALLOWED_LOG_LEVELS || '[]';
+
+type LogLevel = 'INFO' | 'DEBUG' | 'WARN' | 'ERROR';
+
+let LOG_LEVELS: LogLevel[] = ['DEBUG', 'INFO', 'WARN', 'ERROR'];
+try {
+  const parsedLogLevels = JSON.parse(allowedLogLevels);
+  if (parsedLogLevels.length !== 0) {
+    LOG_LEVELS = parsedLogLevels;
+  }
+} catch (e) {
+  console.warn(
+    `[WARN] Could not parse ALLOWED_LOG_LEVELS env variable, logger defaults to all levels`,
+    e
+  );
+}
+
+class Logger {
+  private enabledLevels: Set<LogLevel>;
+
+  constructor(allowedLogLevels: LogLevel[]) {
+    this.enabledLevels = new Set(allowedLogLevels);
+    this.log(`Initialising logger with allowed levels: ${allowedLogLevels.join(', ')}`);
+  }
+
+  log(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('INFO')) {
+      if (typeof message === 'string') {
+        console.log(`[INFO] ${message}`, ...optionalParams);
+      } else {
+        console.log(`[INFO]`, message, ...optionalParams);
+      }
+    }
+  }
+
+  info(message?: any, ...optionalParams: any[]): void {
+    this.log(message, ...optionalParams);
+  }
+
+  debug(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('DEBUG')) {
+      if (typeof message === 'string') {
+        console.debug(`[DEBUG] ${message}`, ...optionalParams);
+      } else {
+        console.debug(`[DEBUG]`, message, ...optionalParams);
+      }
+    }
+  }
+
+  warn(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('WARN')) {
+      if (typeof message === 'string') {
+        console.warn(`[WARN] ${message}`, ...optionalParams);
+      } else {
+        console.warn(`[WARN]`, message, ...optionalParams);
+      }
+    }
+  }
+
+  error(message?: any, ...optionalParams: any[]): void {
+    if (this.enabledLevels.has('ERROR')) {
+      if (typeof message === 'string') {
+        console.error(`[ERROR] ${message}`, ...optionalParams);
+      } else {
+        console.error(`[ERROR]`, message, ...optionalParams);
+      }
+    }
+  }
+}
+
+export const LogHelper = new Logger(LOG_LEVELS);
```

Comment:
I don't think we should export this as `LogHelper`, just `Logger` or better yet just export the individual log methods.

Reply:
I initially went with `Logger` but one of our devDepencencies (concurrenly) has a Logger exported and I don't want people to accidentally import that instead of our customised one. I would like to keep the `LogSomething.log()` syntax to indicate that it is a wrapper around the console. I'm open for naming suggestions, or do you think it would worth the risk of naming it Logger and maybe cause users to accidentally import (or auto-import) the wrong one? 🤔

Reply:
We use concurrently as a devDependency, but since it is in the node_modules folder, almost all import/autoImport tools will default to concurrently because of it's alphabetical precedence.See screenshot

<img width="655" alt="Screenshot 2022-12-05 at 10 27 09" src="https://user-images.githubusercontent.com/13385210/205602001-828990a3-5a79-4cc4-8074-eecfca281ee5.png">

Reply:
We leave it as LogHelper for now, and we'll rename it if we can find a better name that won't try to import from dev dependencies.